### DISCUSSION TITLE
Seeking Guidance on Task Instance Event Handling in Apache Airflow 2.2.5 Plugin

### DISCUSSION THREAD

**Discussion by mishrasangeeta87** [apache/airflow#36790] [Q&A]
Hello everyone,

I'm working on building a plugin for Apache Airflow version 2.2.5, and I need to perform specific actions based on task instance events such as running, success, and failure. Unfortunately, it seems that listener support is not available in Airflow version 2.2.5.

I'm reaching out to the community to inquire if there are alternative approaches or workarounds to achieve this functionality in the absence of listener support. Any insights or suggestions on how I could effectively handle task instance events in Airflow version 2.2.5 would be greatly appreciated.

Thank you in advance for your assistance!

**Comment by potiuk** [comment]
Upgrade to newer Airflow. Seriously, that's the best way.

**Reply by mishrasangeeta87** [reply]
Thanks.
